library/java/net/openid/appauth/IdToken.java

```diff
@@ -160,15 +160,16 @@ private static JSONObject parseJwtSection(String section) throws JSONException {
         return new JSONObject(jsonString);
     }
 
-    static IdToken from(String token) throws JSONException, IdTokenException {
+    public static IdToken from(String token) throws JSONException, IdTokenException {
         String[] sections = token.split("\\.");
 
         if (sections.length <= 1) {
             throw new IdTokenException("ID token must have both header and claims section");
         }
 
-        // We ignore header contents, but parse it to check that it is structurally valid JSON
-        parseJwtSection(sections[0]);
+        // We are using headers to get nonce field from it and also parse it to check that it is
```

Comment:
I'm sorry, I really think it's important to preserve the original comment (because it's coming from the source project) so that it's easier to "work backwards" if we need to in the future.
Something like:
`// (AMN code changes invalidated this comment, 04-2024) <original comment>`
Then:
`// Get header contents because we have observed Azure sending the nonce in the header`

Reply:
done!